### ISSUE TITLE
Possible to make a RunType that uses instanceof?

### ISSUE THREAD

**Issue by breathe** [pelotom/runtypes#22]
Sometimes I'm expecting an instance of a particular class.  Is there a way to make a constraint field based on instanceof check or other typeguard?

I want to express something like this ...

```
class PopulationDistribution { ... }

const Planet = Record({
  type: Literal('planet'),
  location: Vector,
  mass: Number,
  population: InstanceOf(PopulationDistribution),
  habitable: Boolean
})
```

**Comment by pelotom** [comment]
I think it's certainly doable and I'd welcome a PR if you wanted to take a crack at it :)

**Comment by breathe** [comment]
Ok I'll give it a shot!

**Comment by breathe** [comment]
Closed by #23
